doc/developers/cython.rst

```diff
@@ -97,8 +97,6 @@ Tips for performance
 
 * Inline methods and function when it makes sense
 
-* Make sure your Cython compilation units `use NumPy recent C API <https://github.com/scikit-learn/scikit-learn/blob/62a017efa047e9581ae7df8bbaa62cf4c0544ee4/setup.py#L64-L70>`_.
```

Comment:
I'd be interested to have more details about what we need/don't need to do about that before deciding to completely remove it and forget about it. ping @jjerphan, you seem familiar with this stuff :)

Reply:
This is now taken care of natively by `setup.py` (that this might need to be ported to meson):

https://github.com/scikit-learn/scikit-learn/blob/87fa6547d6a864c32851a811a7758837f47bb170/setup.py#L121-L128

so this comment can be removed. :+1:

Reply:
I think `NPY_NO_DEPRECATED_API` must be defined everywhere especially if NumPy 2 is used as old C APIs might be deprecated.

Reply:
Thanks, I think this part has not been ported to Meson, which is likely an oversight ... fixing this is probably doable.

Not 100% sure, but my understanding is that using `NPY_NO_DEPRECATED_API` + `NPY_1_7_API_VERSION` makes sure that we don't use any Numpy C API that is deprecated in Numpy version 1.7.

@jjerphan but do you have a suggestion about which version we should pick? Is `NPY_1_7_API_VERSION` OK? Would using `NPY_1_9_API_VERSION` be better, which is what scipy is doing see https://github.com/scipy/scipy/blob/fffcaa8c58b6909d22063c29e4f9593059a40230/scipy/meson.build#L76-L77?

Reply:
In the meantime I have opened https://github.com/scikit-learn/scikit-learn/pull/29357 to do the same thing as what `setup.py` was doing.

Reply:
I would also pick `NPY_1_9_API_VERSION` to be consistent with SciPy's choice. :+1:

Reply:
#29357 has been merged, so I guess this PR can be merged too?